src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs

```diff
@@ -0,0 +1,462 @@
+            public ManagedObjectWrapperHolder(ManagedObjectWrapper* wrapper)
+            {
+                this.wrapper = wrapper;
+            }
+
+            public unsafe IntPtr ComIp => this.wrapper->As(in ComWrappers.IID_IUnknown);
+
+            ~ManagedObjectWrapperHolder()
+            {
+                Marshal.FreeCoTaskMem((IntPtr)this.wrapper);
+            }
+        }
+
+        internal unsafe struct IUnknownVftbl
+        {
+            public delegate* unmanaged<IntPtr, ref Guid, out IntPtr, int> QueryInterface;
+            public delegate* unmanaged<IntPtr, uint> AddRef;
+            public delegate* unmanaged<IntPtr, uint> Release;
+        }
+
+        /// <summary>
+        /// Globally registered instance of the ComWrappers class for reference tracker support.
+        /// </summary>
+        private static ComWrappers? s_globalInstanceForTrackerSupport;
+
+        /// <summary>
+        /// Globally registered instance of the ComWrappers class for marshalling.
+        /// </summary>
+        private static ComWrappers? s_globalInstanceForMarshalling;
+
+        /// <summary>
+        /// Create a COM representation of the supplied object that can be passed to a non-managed environment.
+        /// </summary>
+        /// <param name="instance">The managed object to expose outside the .NET runtime.</param>
+        /// <param name="flags">Flags used to configure the generated interface.</param>
+        /// <returns>The generated COM interface that can be passed outside the .NET runtime.</returns>
+        /// <remarks>
+        /// If a COM representation was previously created for the specified <paramref name="instance" /> using
+        /// this <see cref="ComWrappers" /> instance, the previously created COM interface will be returned.
+        /// If not, a new one will be created.
+        /// </remarks>
+        public unsafe IntPtr GetOrCreateComInterfaceForObject(object instance, CreateComInterfaceFlags flags)
+        {
+            if (instance == null)
+                throw new ArgumentNullException(nameof(instance));
+
+            if (CCWTable.TryGetValue(instance, out ManagedObjectWrapperHolder ccwValue))
+            {
+                return ccwValue.ComIp;
+            }
+
+            ManagedObjectWrapperHolder newValue = CCWTable.GetValue(instance, (c) =>
+            {
+                ManagedObjectWrapper* value = CreateCCW(this, c, flags);
+                return new ManagedObjectWrapperHolder(value);
+            });
+            return newValue.ComIp;
+        }
+
+        private static unsafe ManagedObjectWrapper* CreateCCW(ComWrappers impl, object instance, CreateComInterfaceFlags flags)
+        {
+            ComInterfaceEntry* userDefined = impl.ComputeVtables(instance, flags, out int userDefinedCount);
+
+            // Maximum number of runtime supplied vtables.
+            Span<ComInterfaceEntry> runtimeDefinedLocal = stackalloc ComInterfaceEntry[4];
```

Comment:
```suggestion
            Span<IntPtr> runtimeDefinedLocal = stackalloc IntPtr[4];
```
You only need the Vtable field now, so it can be just Span of IntPtrs.